File: lib/models/nutrition.dart
class NutritionLog {
  final int id;
  final DateTime logDate;
  final double calories;
  final double proteinG;
  final double carbsG;
  final double fatG;
  final double waterMl;
  final String? notes;

  NutritionLog({
    required this.id,
    required this.logDate,
    required this.calories,
    required this.proteinG,
    required this.carbsG,
    required this.fatG,
    required this.waterMl,
    this.notes,
  });

  factory NutritionLog.fromJson(Map<String, dynamic> json) => NutritionLog(
    id: json['id'],
    logDate: DateTime.parse(json['log_date']),
    calories: (json['calories'] as num).toDouble(),
    proteinG: (json['protein_g'] as num).toDouble(),
    carbsG: (json['carbs_g'] as num).toDouble(),
    fatG: (json['fat_g'] as num).toDouble(),
    waterMl: (json['water_ml'] as num).toDouble(),
    notes: json['notes'],
  );
}

File: legacy/backend_fastapi/app/schemas/nutrition.py
from datetime import date

from pydantic import BaseModel, ConfigDict, Field


class NutritionLogUpsert(BaseModel):
    log_date: date
    calories: float = Field(default=0, ge=0)
    protein_g: float = Field(default=0, ge=0)
    carbs_g: float = Field(default=0, ge=0)
    fat_g: float = Field(default=0, ge=0)
    water_ml: float = Field(default=0, ge=0)
    notes: str | None = None


class NutritionLogResponse(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: int
    log_date: date
    calories: float
    protein_g: float
    carbs_g: float
    fat_g: float
    water_ml: float
    notes: str | None
